### PR TITLE
Add Guide/Feature: Colour Generator

### PR DESCRIPTION
**Pull request by ShaunR1991**
# Description

This pull request introduces a new feature to the SplashKit guides - tools for generating and selecting colours. The addition includes two key components:

1. **Colour Generator**: A tool for visualising and generating precise RGB(A) and HSB colour values, with immediate visual feedback and corresponding SplashKit functions in C++, C#, and Python.
2. **Colour Palette**: An interactive tool showing the full range of SplashKit's predefined colours, allowing users to visually select a colour and see how to use it in their projects.

These tools enhance the user experience by simplifying colour manipulation and providing a seamless integration of SplashKit colours into projects.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation (update or new)

## How Has This Been Tested?

- Verified the functionality of the **Colour Generator** by adjusting RGB(A) and HSB sliders, observing the live preview and generated SplashKit code.
- Interacted with the **Colour Palette** to select colours and validated that the correct code snippets are displayed in the tabs.
- Tested both features in the following environments:
  - Latest Chrome
  - Latest Firefox
- Ensured the build and preview commands (`npm run build` and `npm run preview`) executed successfully without errors.

## Testing Checklist

- [x] Tested in latest Chrome
- [x] Tested in latest Firefox
- [x] npm run build
- [x] npm run preview

### If involving code

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.

### If modified config files

- [x] I have checked the following files for changes:
  - [x] `astro.config.mjs`

## Folders and Files Added/Modified

- **Added**:
  - `scripts/splashkit-color-palette.json`: JSON file defining the SplashKit colour data.
  - `src/components/ColorGenerator.astro`: Component for the Colour Generator.
  - `src/components/ColorPalette.astro`: Component for the Colour Palette.
  - `src/content/docs/guides/Color/0-colour-generator.mdx`: Documentation for the Colour tools.
  - `src/content/docs/guides/Color/images/colour_wheel.png`: Colour wheel image for the HSB section.

- **Modified**:
  - `astro.config.mjs`: Updated to include new components and pages.

## Additional Notes

While the current implementation updates code blocks by overwriting their content with plaintext, there is room for extension. Future iterations could fully integrate a method to dynamically format code blocks based on the selected language, maintaining proper syntax highlighting instead of plain text replacement. This would enhance readability and provide a more polished experience for users.

### CODE REVIEW

**Review by EthanRin — APPROVED** [review]
### Overview
I've reviewed this PR and I absolutely love the idea of this guide. The guide itself is well-structured and informative. This interactive approach also allowed users to test out different colors and HSB in real-time as well as getting to know all of the color available on SplashKit through the click of a button.

### Codes 
The new components are well-structured with plenty of comments to help improve reability. 

### Json Files
The JSON files for the color is well-structured and consistent; although a possible guide to on how to add on to the JSON file would be good.

 I've also pulled this PR on to my local fork and had no issues running the build command.

File: src/components/ColorGenerator.astro [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+  }
+
+  /* Individual styling for Red slider */
+  #red-slider {
+    background-color: #f8d7d7; /* Light red background */
+  }
+  #red-slider:hover {
+    background-color: red; /* Intense red on hover */
+  }
+  #red-slider::-webkit-slider-thumb {
```
Comment (ctrlvish) [inline src/components/ColorGenerator.astro:32]: Hi @ShaunR1991 , this is a wonderful PR and addition to the documentation site. It gives users the ability to quickly select the splashkit colours from the color palette and generate colors from the generator. I've reviewed the guide and it is well written too.

I have one change to suggest here on this style for the `#red-slider::-webkit-slider-thumb`, it uses custom styles that aren't consistent with the other slider thumb styles. Also the other slider thumbs for blue and green, there are `background-color` properties being changed but not actually applied as its missing the `-webkit-appearance: none; /* Remove default appearance */` style to remove the default styles first.

Here the red slider thumb has custom styles and the other slider thumbs have default styles
image

File: src/components/ColorGenerator.astro [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+  }
+
+  /* Styling for Green slider */
+  #green-slider {
+    background-color: #d7f8d7; /* Light green */
+  }
+  #green-slider:hover {
+    background-color: green;
+  }
+  #green-slider::-webkit-slider-thumb {
```
Comment (ctrlvish) [inline src/components/ColorGenerator.astro:56]: I would suggest adding this line here to use the custom color
`webkit-appearance: none;`

File: src/components/ColorGenerator.astro [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+  }
+
+  /* Styling for Blue slider */
+  #blue-slider {
+    background-color: #d7d7f8; /* Light blue */
+  }
+  #blue-slider:hover {
+    background-color: blue;
+  }
+  #blue-slider::-webkit-slider-thumb {
```
Comment (ctrlvish) [inline src/components/ColorGenerator.astro:71]: same here:
`webkit-appearance: none;`

**Review by ctrlvish — CHANGES_REQUESTED** [review]

**Review by omckeon — APPROVED** [review]
I've made some adjustments to the styling of the interactive elements (slider coloring and text color on the color palette) and have separated the color palette code blocks for easier copying. The code parts are still missing highlighting, but now it keeps the same general formatting.

I'm very excited to be able to use this now!

File: src/components/ColorGenerator.astro (hunk 1 above)
Comment (omckeon) [reply src/components/ColorGenerator.astro:30]: Great catch @ctrlvish ! I was able to use this information when finalising this PR :)

File: src/components/ColorGenerator.astro (hunk 2 above)
Comment (omckeon) [reply src/components/ColorGenerator.astro:40]: Great catch @ctrlvish ! I was able to use this information when finalising this PR merge :)

File: src/components/ColorGenerator.astro (hunk 3 above)
Comment (omckeon) [reply src/components/ColorGenerator.astro:50]: Great catch @ctrlvish ! I was able to use this information when finalising this PR :)

**Pull request merged**